### PR TITLE
Implementing license header checking in CI

### LINKED ISSUE #182: Implement license header checking in CI
**Is your feature request related to a problem? Please describe.**
Files without licenses should automatically be detected.

**Describe the solution you'd like**
CI job for checking if all python and cpp files include license 

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by john-science**
## Purpose

- To implement license header checking in CI

## Proposed Changes

What does this PR add, remove or fix?

I have added one GitHub Action CI (to check that every Python file has a Copyright header at the top). And I had to add one metadata file to the root of the repo, to make this CI action run properly.

## Issues

- To close #182 

## Testing

- How was it tested, what were the results?

This is it's own CI, so it is self-testing. After running it the first time, I found three `__init__.py` files that were lacking the header. Kind of an oddity, but looking around at some other open-source projects that have this license header rule (e.g. [armi](https://github.com/terrapower/armi/), this appears to be the standard, so I went with it.

### CODE REVIEW

**Comment by maciejmajek** [comment]
Hi, Thank you! I don't think this is mentioned in the readmes, we use pre-commit for code formatting. Please run the following:
```bash
pre-commit run -a
```

**Comment by adamdbrw** [comment]
Thank you for this contribution!

**Comment by john-science** [comment]
> Hi, Thank you! I don't think this is mentioned in the readmes, we use pre-commit for code formatting. Please run the following:
> 
> ```shell
> pre-commit run -a
> ```

Oh, sorry about that.

Okay, I ran `precommit run -a` and it came back clean.  (After I fixed the linting errors.)

File: .licenserc.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "**/*.py": "# Copyright "
```
Comment (adamdbrw) [inline .licenserc.json]:
```suggestion
  "**/*.py": [
  "# Copyright ",
  "#",
  "# Licensed under the Apache License, Version 2.0 (the \"License\");"
  ],
  "**/*.{h,cpp,hpp}": [
  "// Copyright ", 
  "//",
  "// Licensed under the Apache License, Version 2.0 (the \"License\");"
  ]
```

Comment (adamdbrw) [reply .licenserc.json]: Rationale: 
We have C++ as well as Python in the project
We should not allow to include non-Apache Licenses in our core (we can add 3rd party src folder and exclude it from this requirement to allow for compatible licenses such as MIT, BSD 3, Mozilla later).

**Review by adamdbrw — COMMENTED** [review]
Checker suggestions

File: .licenserc.json (hunk 1 above)
Comment (maciejmajek) [reply .licenserc.json]: I've tested it using `npx @kt3k/license-checker`
with config
```json
{
    "**/*.py": [
        "# Copyright ",
        "#",
        "# Licensed under the Apache License, Version 2.0 (the \"License\");"
    ],
    "**/*.{h,cpp,hpp}": [
        "// Copyright ",
        "//",
        "// Licensed under the Apache License, Version 2.0 (the \"License\");"
    ]
}
```
and it seems to be working as expected

Comment (adamdbrw) [reply .licenserc.json]: Great! How about you either apply + modify my suggestion or submit your patch (tested) with my co-authorship? I guess there are whitespace differences.

Comment (maciejmajek) [reply .licenserc.json]: Up to @john-science

Comment (john-science) [reply .licenserc.json]: Yeah, that change is solid. Taken.

I was only looking at the Python code, my bad.

**Review by maciejmajek — COMMENTED** [review]
Please run the pre-commit 🙇🏼‍♂️, otherwise LGTM

**Comment by john-science** [comment]
> Please run the pre-commit 🙇🏼‍♂️, otherwise LGTM

Okay, linter came back clean.

**Review by maciejmajek — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**